lib/Runtime/Library/WebAssemblyModule.cpp

```diff
@@ -219,12 +219,13 @@ WebAssemblyModule::CreateModule(
     {
         Js::AutoDynamicCodeReference dynamicFunctionReference(scriptContext);
         SRCINFO const * srcInfo = scriptContext->cache->noContextGlobalSourceInfo;
-        Js::Utf8SourceInfo* utf8SourceInfo = Utf8SourceInfo::New(scriptContext, (LPCUTF8)buffer, lengthBytes / sizeof(char16), lengthBytes, srcInfo, false);
 
         // copy buffer so external changes to it don't cause issues when defer parsing
-        byte* newBuffer = RecyclerNewArray(scriptContext->GetRecycler(), byte, lengthBytes);
+        byte* newBuffer = RecyclerNewArrayLeaf(scriptContext->GetRecycler(), byte, lengthBytes);
         js_memcpy_s(newBuffer, lengthBytes, buffer, lengthBytes);
 
+        Js::Utf8SourceInfo* utf8SourceInfo = Utf8SourceInfo::NewWithNoCopy(scriptContext, (LPCUTF8)newBuffer, lengthBytes / sizeof(char16), lengthBytes, srcInfo, false);
```

Comment:
I checked a little closer at the `Utf8SourceInfo` constructors and I am wondering now if we should use `Utf8SourceInfo::New` and then get its copy of the buffer instead of making the copy ourselves.